config_parser.cc

```diff
@@ -199,6 +207,7 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
           new_config);
       config_stack.push(new_config);
     } else if (token_type == TOKEN_TYPE_END_BLOCK) {
+      balanced += -1;
```

Comment:
Also idiomatic to say `balanced--;` or even `balanced -= 1;` rather than `balanced += -1`.